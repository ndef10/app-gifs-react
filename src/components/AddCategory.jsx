import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {    
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {        
        setInputValue(target.value);
    }

    const onSubmit = ( event ) => {
        // console.log('Hola mundo desde Submit!!!!!!!!!')
        event.preventDefault();    
        if(inputValue.trim().length <= 1) return;       
        onNewCategory(inputValue);
        setInputValue('');
    }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
        <input
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange= { onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,   
  
}