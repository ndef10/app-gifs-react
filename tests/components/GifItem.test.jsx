import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components";


describe('prueba en <GifItem/>', () => {

    const title = 'mono';
    const url = 'http://jhdhdhdhhdhdhhd.com/gato';

    test('GiItem debe hacer match con el snapshot', () => {
        
        const { container } = render ( <GifItem title = { title } url = { url } /> );

        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el atl indicado', () => {

        render ( <GifItem title ={ title } url ={ url } /> );

        // screen.debug(); 
        // expect( screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );


    });

    test('debe mostrar el titulo en el componente', () => {

        render ( <GifItem title ={ title } url ={ url } /> );
        expect( title ).toBeTruthy( );



    })

})