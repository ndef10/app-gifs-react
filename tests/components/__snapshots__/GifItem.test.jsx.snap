// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en <GifItem/> GiItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="mono"
      src="http://jhdhdhdhhdhdhhd.com/gato"
    />
    <p>
      mono
    </p>
  </div>
</div>
`;
